pkg/bugreport/controlplane.go

```diff
@@ -21,26 +21,41 @@ func (c *Config) collectControlPlaneLogs() error {
 			),
 		})
 	if err != nil {
-		return fmt.Errorf("error getting control plane pods: %w", err)
+		c.completionFailure("Error getting control plane pods: %s", err)
+		goto afterPodLogCollection
```

Comment:
Depending on how we feel about `goto`, one alternative I can think of is to put the `for` loop after this in an `else` block.

Reply:
I prefer this to avoid unnecessary nesting.

Reply:
+1 an else block or a helper function is my vote here...maybe even a `defer`?

Reply:
I think this is a pretty reasonable use of `goto`, which of course is also used in many Go std libs. I don't see a good reason to: 1. increase the nesting by moving this into an else block, 2. use a defer just to avoid goto.

Reply:
It's not a bad use of go to; agreed. However, I feel like the use of goto here is equivalent to a finally block in a Java based language, and `defer` feels more appropriate semantically assuming that's the case. It's up to you though; this is not a hill I'll die on